File: Computabilidade-e-Complexidade-de-Algoritmos-main/golang/21.go
package main
import "fmt"

func aAposCadaB(s string) bool {
    for i := 0; i < len(s)-1; i++ {
        if s[i] == 'b' && s[i+1] != 'a' {
            return false
        }
    }
    return true
}

func main() {
    fmt.Println(aAposCadaB("baa"))   // False
    fmt.Println(aAposCadaB("abab"))  // True
}

File: Computabilidade-e-Complexidade-de-Algoritmos-main/java/21.java
public class AFNAposCadaB {
    public static boolean aAposCadaB(String s) {
        for (int i = 0; i < s.length() - 1; i++) {
            if (s.charAt(i) == 'b' && s.charAt(i + 1) != 'a') {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        System.out.println(aAposCadaB("baa"));   // False
        System.out.println(aAposCadaB("abab"));  // True
    }
}
